src/Albany_Application.cpp

```diff
@@ -1996,8 +1993,15 @@ Application::computeGlobalTangent(
   if (Teuchos::nonnull(params)) {
     TanFadType p;
     int        num_cols_tot = param_offset + num_cols_p;
+
+    TEUCHOS_TEST_FOR_EXCEPTION(
+      num_cols_tot >  tangent_deriv_dim,
```

Comment:
Actually, I now see that `param_offset` is not the offset of the current parameter, but rather 0 or num_cols_x, depending on `sum_derivs`. However, I wonder how this can work if `sum_derivs=false`, since `calcTangentDerivDimension` (which computes `tangent_deriv_dim`)  does not seem to consider this. So I am afraid that, if `sum_derivs=false`, we would trigger this exception.

Reply:
It actually seems that `sum_derivs=false` always. So now I'm confused.

Reply:
I think `offset` is always zero because `sum_derivs` it's always `false`, we should clean this up at some point. so `num_cols_tot` should always be less or equal to `tangent_deriv_dim`.

Reply:
If `sum_derivs=false`, then `param_offset=num_cols_x`. I guess we don't really ever use Vx/Vxdot/Vxdotdot then, otherwise the offset would be nonzero...

Reply:
Btw, if `sum_derivs=true`, then we need to ensure that the fad length is >=max(num_cols_p, num_cols_x).

I agree, this needs some cleanup at some point. I'm very wary of code that we may be not using, and might be super buggy.

Reply:
Oh right sorry, yes you are right. If I remember correctly Vx/Vxdot/Vxdotdot  are needed for some software optimization, but we never use them and we probably should clean up the code at some point, maybe after your PR.